consensus/src/common.rs

```diff
@@ -120,21 +121,16 @@ impl Chain<ProtocolState> for ProtocolStateChain {
         self.0.len()
     }
 
-    fn last_vrf_hash(&self) -> Option<String> {
+    fn last_vrf_hash(&self) -> Option<Fp> {
         self.top().map(|s| {
-            s.body
-                .consensus_state
-                .last_vrf_output
-                .hash()
-                .as_ref()
-                .encode_hex::<String>()
+            let mut hasher = create_kimchi::<VrfOutputTruncated>(());
+            hasher.hash(&s.body.consensus_state.last_vrf_output)
         })
     }
 
-    // FIXME: this currently returns a blake2b hash. It should
-    // return a poseidon hash according to: https://github.com/MinaProtocol/mina/blob/32f529d8d8d712a44ee75be66061ce08cbdc8924/docs/specs/consensus/README.md#517-hashstate
-    fn state_hash(&self) -> Option<StateHash> {
-        self.top().map(|s| s.hash())
+    fn state_hash(&self) -> Option<Fp> {
+        let mut hasher = create_kimchi::<ProtocolState>(());
```

Comment:
It's expensive to create a new hasher every time, better to create a pool and reuse the hasher.